src/Components/test/testassets/TestServer/Startup.cs

```diff
@@ -96,6 +105,13 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env)
             {
                 endpoints.MapControllers();
                 endpoints.MapRazorPages();
+
+                // Redirect for convenience when testing locally since we're hosting the app at /subdir/
+                endpoints.Map("/", context =>
+                {
+                    context.Response.Redirect("/subdir");
+                    return Task.CompletedTask;
```

Comment:
Another *nice improvement* that I wanted to do. I always get upset when I launch the app and get a 404.

Reply:
This doesn't seem quite right to me. If there's a request for `/_framework/something.js` I'd expect it to return a 404, not a redirection to `/subdir`.

Perhaps a better fix would be changing the `launchUrl` in `launchSettings.json`.

Reply:
This is using routing, so it's specifically going to match requests to the root URL `/` and only `/`

Reply:
Map with middleware = map a prefix | Map with routing = match exactly